igcollect/linux_disklatency.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+"""igcollect - Linux Disk I/O latency
+
+Copyright (c) 2019 InnoGames GmbH
+"""
+
+from argparse import ArgumentParser
+from subprocess import check_output
+from time import time
+import libvirt, sys
+
+def main():
+    URI = "qemu:///system"
+    strpoolname = 'xen-data'
+    strvolname = 'test-vol'
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='None')
+    args = parser.parse_args()
+    template = args.prefix + '.{}.{} {} ' + str(int(time()))
+
+    try:
+        conn = libvirt.open(URI)
+    except:
+        print('Failed to open connection to qemu:///system', file=sys.stderr)
+        exit(1)
+
+    try:
+        strgpool = conn.storagePoolLookupByName(strpoolname)
+    except:
+        print('Failed to find storage pool ' + strpoolname, file=sys.stderr)
+        exit(1)
+
+    try:
+        stgvol = strgpool.storageVolLookupByName(strvolname)
+        stgvolpath = stgvol.path()
+
+        output_read = check_output(["/usr/bin/sudo", "/usr/bin/ioping", "-BD",
+                                    "-c10", "-i0.1", stgvolpath],
+                                   universal_newlines=True)
+
+        output_write = check_output(["/usr/bin/sudo", "/usr/bin/ioping", "-BDWWW",
+                                     "-c10", "-i0.1", stgvolpath],
+                                    universal_newlines=True)
+
+        output = {
+            'read'  : output_read,
+            'write' : output_write,
+        }
+
+        for mode, data in output.items():
+            measurement = data.split()
+
+            results = {
+                'min' : measurement[4],
+                'avg' : measurement[5],
+                'max' : measurement[6],
+            }
+
+            for k, v in results.items():
+                print(template.format(mode, k, int(v)))
+
+    except:
```

Comment:
Same or rather why at all. If it dies because, let's say, parsing errors, that is an error you want to display to user.

Reply:
changed